src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(anotherAnimal) {
+    if (anotherAnimal instanceof Herbivore && !anotherAnimal.hidden) {
+      anotherAnimal.health -= 50;
+      Animal.alive = Animal.alive.filter((enimal) => enimal.health > 0);
```

Comment:
Let's do this only if we know that there are animals that has less or equal 0 health